spyder/plugins/findinfiles/widgets/main_widget.py

```diff
@@ -759,6 +763,37 @@ def set_max_results(self, value=None):
         else:
             self.set_conf('max_results', value)
 
+    def show_error(self, error_msg, exclude):
+        """Show a regexp error message with an icon."""
+        self._show_icon_message(extra_info=error_msg, exclude=exclude)
+
+    def _show_icon_message(self, extra_info=None, exclude=False):
+        """
+        Show a message to users with an icon when there's an error in the passed regexp.
```

Comment:
```suggestion
    # ---- Private API
    # ------------------------------------------------------------------------
    def _show_error(self, error_msg, exclude):
        """
        Show a regexp error message with an icon.
```

Make method private, remove extra method and add header section.